src/DependencyEmitter/UsesDependencyEmitter.php

```diff
@@ -34,4 +37,21 @@ public function applyDependencies(AstMap $astMap, Result $dependencyResult): voi
             }
         }
     }
+
+    protected function isNamespace(AstDependency $dependency, AstMap $astMap): bool
+    {
+        $dependencyFQDN = $dependency->getTokenName()->toString();
+
+        foreach ($astMap->getAstClassReferences() as $class) {
```

Comment:
You might want to do a same type of for each loop for functions here as well.

Reply:
You're right, nice catch :)